data-prepper-plugins/anomaly-detector-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/anomalydetector/AnomalyDetectorProcessorTests.java

```diff
@@ -0,0 +1,139 @@
+        }
+        anomalyDetectorProcessor.doExecute(records);
+        
+        final List<Record<Event>> recordsWithAnomaly = (List<Record<Event>>) anomalyDetectorProcessor.doExecute(Collections.singletonList(getLatencyMessage(UUID.randomUUID().toString(), ThreadLocalRandom.current().nextDouble(10.4, 10.8))));
+        assertThat(recordsWithAnomaly.size(), equalTo(1));
+        Event event = recordsWithAnomaly.get(0).getData();
+        List<Double> deviation = event.get(AnomalyDetectorProcessor.DEVIATION_KEY, List.class);
+        assertThat((double)deviation.get(0), greaterThan(9.0));
+        double grade = (double)event.get(AnomalyDetectorProcessor.GRADE_KEY, Double.class);
+        assertThat(grade, equalTo(1.0));
+    }
+
+    @Test
+    void testAnomalyDetectorProcessorTwoKeys() {
+        List<String> keyList = new ArrayList<String>();
+        keyList.add("latency");
+        keyList.add("bytes");
+        when(mockConfig.getKeys()).thenReturn(keyList);
+        RandomCutForestModeConfig randomCutForestModeConfig = new RandomCutForestModeConfig();
+        AnomalyDetectorMode anomalyDetectorMode = new RandomCutForestMode(randomCutForestModeConfig);
+
+        when(mockConfig.getDetectorMode()).thenReturn(modeConfiguration);
+        when(modeConfiguration.getPluginName()).thenReturn(UUID.randomUUID().toString());
+        when(modeConfiguration.getPluginSettings()).thenReturn(Collections.emptyMap());
+
+        when(pluginFactory.loadPlugin(eq(AnomalyDetectorMode.class), any(PluginSetting.class)))
+                .thenReturn(anomalyDetectorMode);
+        anomalyDetectorProcessor = new AnomalyDetectorProcessor(mockConfig, pluginMetrics, pluginFactory);
+        final int numSamples = 1024;
+        final List<Record<Event>> records = new ArrayList<Record<Event>>();
+        for (int i = 0; i < numSamples; i++) {
+            records.add(getLatencyBytesMessage(UUID.randomUUID().toString(), ThreadLocalRandom.current().nextDouble(0.5, 0.6), ThreadLocalRandom.current().nextLong(100, 110)));
+        }
+        anomalyDetectorProcessor.doExecute(records);
+        final List<Record<Event>> recordsWithAnomaly = (List<Record<Event>>) anomalyDetectorProcessor.doExecute(Collections.singletonList(getLatencyBytesMessage(UUID.randomUUID().toString(), ThreadLocalRandom.current().nextDouble(10.4, 10.8), ThreadLocalRandom.current().nextLong(1000, 1110))));
+        assertThat(recordsWithAnomaly.size(), equalTo(1));
+        Event event = recordsWithAnomaly.get(0).getData();
+        List<Double> deviation = event.get(AnomalyDetectorProcessor.DEVIATION_KEY, List.class);
+        for (int i = 0; i < keyList.size(); i++) {
+            assertThat((double)deviation.get(i), greaterThan(9.0));
+        }
+        double grade = (double)event.get(AnomalyDetectorProcessor.GRADE_KEY, Double.class);
+        assertThat(grade, equalTo(1.0));
+    }
+
+}
```

Comment:
This file is included in the PR. What is it for?

```
data-prepper-plugins/anomaly-detector-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/anomalydetector/modes/.RandomCutForestModeTests.java.swp
```

Reply:
That was a mistake. Will remove it.